packages/lib/pod/src/pod.ts

```diff
@@ -164,4 +169,49 @@ export class POD {
       parsedPOD.signerPublicKey
     );
   }
+
+  /**
+   * Converts this POD to a JSON-compatible format which can be safely
+   * serialized using `JSON.stringify` without any loss of information.  To
+   * reconstitute a POD object from JSON, see {@link fromJSON}.
+   *
+   * @returns a JSON-compatible representation of this POD.
+   */
+  public toJSON(): JSONPOD {
+    return {
+      entries: podEntriesToJSON(this._content.asEntries()),
+      signature: this._signature,
+      signerPublicKey: this._signerPublicKey
+    };
+  }
+
+  /**
+   * Rebuilds a POD object from the JSON-compatible format produced by
+   * {@link toJSON}.  The input can be taken directly from `JSON.parse` and
+   * will be fully validated by this function.
+   *
+   * @param jsonPOD the JSON-encoded POD.
+   * @returns a new POD object
+   * @throws TypeError if the input is malformed
+   * @throws RangeError if a value is outside of the bounds
+   */
+  public static fromJSON(jsonPOD: JSONPOD): POD {
+    requireType("jsonPOD", jsonPOD, "object");
+
+    // Rule out unrecognized keys to avoid lossy parsing of a future format
+    // which might have more fields.  This also implicitly rejects a "version"
+    // field, so that if we add one later old code won't ignore it.
+    if (Object.keys(jsonPOD).length !== 3) {
```

Comment:
Strictly speaking, this could fail if we re-organized the keys such that there were still three of them, but with different names. I'm vaguely aware that POD2 treats `signerPublicKey` as an entry, so I could imagine an object with "entries", "signature", and "version", where version = 2.

Reply:
The names are being checked later on.  Each of the names accessed in the POD.load call below is checked against undefined.

So yeah, this line is only checking the number of keys.  The specific names of the keys are in the error message to be helpful, but isn't being checked by this line.  I could put that in a comment.